### PR TITLE
Fix positions column in candidates page

### PR DESCRIPTION
**Pull request by veresdavid**
[Trello card](https://trello.com/c/7BStKyCW)

### CODE REVIEW

File: web/src/test/resources/com/epam/rft/atsy/cucumber/Welcome.feature [hunk 1]
```diff
@@ -2,6 +2,7 @@ Feature: Welcome
 
   Background:
     Given The user signed in
+    Given The Applications of the candidate with id 1 are deleted
```
Comment (kbertalan) [inline web/src/test/resources/com/epam/rft/atsy/cucumber/Welcome.feature]: If you come back to this feature file after few months and you see this line, then you will have no idea which is the candidate with id 1.

I suggest a more readable sentence here, like:

```
Given The following application positions exist for candidates:
    | candidate   | position  |
    | Candidate A |           |
    | Candidate B |           |
    | Candidate C | Fejlesztő |
```

In this case your java code can make sure that the A and B candidates have no applications, but candidate C has one with position `Fejlesztő`.
I suggest to rework the temporary endpoint to support this functionality.

File: web/src/main/webapp/resources/js/atsy-candidate.js [hunk 2]
```diff
@@ -51,3 +69,5 @@ function actionFormatter(value, row, index) {
 function positionFormatter(value) {
```
Comment (lamfalusy) [inline web/src/main/webapp/resources/js/atsy-candidate.js:69]: This function is not used after this commit, it should be deleted or it should be used as I suggested above.

File: service/src/main/java/com/epam/rft/atsy/service/impl/StatesHistoryServiceImpl.java [hunk 3]
```diff
@@ -62,6 +63,17 @@
...
+  public void deleteStateHistoriesByApplication(ApplicationDTO applicationDTO) {
+    List<StatesHistoryEntity>
+        statesHistoryEntities =
+        statesHistoryRepository.findByApplicationEntityOrderByCreationDateDesc(
+            applicationsRepository.getOne(applicationDTO.getId()));
+
+    statesHistoryRepository.delete(statesHistoryEntities);
+  }
+
+
```
Comment (lamfalusy) [inline service/src/main/java/com/epam/rft/atsy/service/impl/StatesHistoryServiceImpl.java]: Delete this empty line.

File: service/src/test/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImplTest.java [hunk 4]
```diff
@@ -180,6 +201,42 @@ public void getApplicationByCandidateDTOShouldReturnThreeElementListOfApplicatio
...
+    // When
+    applicationsService.deleteApplicationsByCandidateDTO(candidateDTO);
+
+    // Then
+    verify(statesHistoryService, atLeastOnce()).deleteStateHistoriesByApplication(applicationDTO);
+    verify(applicationsRepository, atLeastOnce()).delete(applicationEntity);
+
+  }
+
+
```
Comment (lamfalusy) [inline service/src/test/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImplTest.java:239]: Delete this empty line.

File: service/src/main/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImpl.java [hunk 5]
```diff
@@ -26,6 +30,25 @@
   @Autowired
   private ConverterService converterService;
 
+  @Autowired
+  private CandidateRepository candidateRepository;
+
+
```
Comment (lamfalusy) [inline service/src/main/java/com/epam/rft/atsy/service/impl/ApplicationsServiceImpl.java]: Delete this empty line.

File: service/src/main/java/com/epam/rft/atsy/service/converter/impl/CandidateOneWayConverter.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    Set<String> positions = new HashSet<>();
+
+    List<ApplicationDTO>
+        applicationDTOs =
+        applicationsService.getApplicationsByCandidateDTO(candidateDTO);
+
+    for (ApplicationDTO applicationDTO : applicationDTOs) {
+
+      PositionDTO positionDTO = positionService.getPositionById(applicationDTO.getPositionId());
```
Comment (lamfalusy) [inline service/src/main/java/com/epam/rft/atsy/service/converter/impl/CandidateOneWayConverter.java]: You collect PositionDTOs one by one, what result in a lot of database operation.

It is possible to eliminate unnecessary database selects if you collect PositionDTOs in one call. In JpaRepository there is a method: List<T> findAll(Iterable< ID > var1);. In this situation it seems perfect, refactor getPositionById method and use findAll.

File: service/src/test/java/com/epam/rft/atsy/service/converter/CandidateOneWayConverterTest.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+    // When
+    CandidateDTO result = candidateOneWayConverter.firstTypeToSecondType(candidateEntity);
+
+    // Then
+    assertThat(result, equalTo(candidateDTO));
+
+  }
+
+
```
Comment (lamfalusy) [inline service/src/test/java/com/epam/rft/atsy/service/converter/CandidateOneWayConverterTest.java]: Delete this duplicated empty line.

File: web/src/main/webapp/WEB-INF/pages/candidates.jsp [hunk 8]
```diff
@@ -51,7 +51,7 @@
                             </th>
                             <th data-field="phone" data-align="left" data-sortable="true"><spring:message
                                     code="welcome.candidates.table.phone.title"/></th>
-                            <th data-field="positions" data-align="left" data-sortable="true" data-formatter="positionFormatter">
```
Comment (lamfalusy) [inline web/src/main/webapp/WEB-INF/pages/candidates.jsp]: I like this data-formatter, what was wrong with it?

Without this the position names are concatenated without space.

**Pull request merged**